### ISSUE TITLE
improve caching?

### ISSUE THREAD

**Issue by paulirish** [jjNford/github-repositories#17]
Love this extension. So great

I'm wondering.. 

I'm on 3g so the pull of all the repo information takes a good 15 seconds. Can the localStorage cache be pulled up and used.. and then the view changed if the API call returns anything different repos (basically highly unlikely)

I imagine changed watcher/fork counts wouldnt demand a redraw.

And then the new response goes in the cache. 

Or something like that?  

Basically I want me some more speed!! :D

**Comment by jjNford** [comment]
Awesome Paul, glad to hear!  Great ideas, I should have some down time this weekend to look into these issues.  Both should be doable.  Thanks for the input.

**Comment by jjNford** [comment]
Hey Paul, haven't forgot about this.  

Been swamped at work, working on a rewrite for this after chance i get, coded this really fast just to get it up but it has had enough of a response that I need to fix the code up so its easier to add to.  Shouldn't be too long before the new stuff is up with your requests added in.

**Comment by paulirish** [comment]
it's cool

a quickish fix might be using https://github.com/paulirish/jquery-ajax-localstorage-cache

and you'd have to change your getJSON to $.ajax but it would be pretty ez-ish

:)

**Comment by jjNford** [comment]
Very convenient repo but not sure if it would work well with the GitHub API. The API only returns 40 repos at a time, a separate request is send out to retrieve forked repos data (to get parent name). Thus I am recursively calling the interface to pull repos. 

But no worries I have a solution to make this work perfectly for all cases (any data speed) where repos will be dynamically added to the DOM and cached in the background when the popup is not opened. This way the initial load is quick and expired cache is shown and updated dynamically. 

The code has been refactor and I have most of the new caching stuff written. Shouldn't be much longer. Sorry been gone a ton for work :P

Also fixed CSS issues that existed.

**Comment by jjNford** [comment]
hey paul... tried a few different things for caching (i have totally not been ignoring this :P).  think i have a solution.  i have implemented it for the repos (other section are not implemented yet from refactor).  the changes are on the dev branch, if you would like to test it out that would be great.

**Comment by paulirish** [comment]
just caught up to this. looks like you're about to finish a release. :)

excited!

On Tue, Feb 28, 2012 at 10:28 PM, JJ Ford <
reply@reply.github.com

**Comment by jjNford** [comment]
yep yep, sorry it took so long... this was kind of a learn javascript build something project haha.  i am refactoring right now.  i pulled down the dev branch so nothing for you to test... but... i'm about to push up a version that will be testable.  it is much faster, verified this using an Edge connection :)

**Comment by paulirish** [comment]
nice work, sir!

really excited for it.

**Comment by jjNford** [comment]
I think you will be happy with the new stuff.  All the loading and displaying elements are complete.  Just finishing up some details and I'll upload the release.  Would love if you could spread the word, would like to see this thing get some traffic!

**Comment by jjNford** [comment]
Hey Paul, just pushed newest version.  However it is not optimized (concat, minified, img opts).  I was trying to use the html5bp build script but had trouble getting a few things working.  Was hoping you could take a look at it if you get some free time.  There are a few .js files that can't be concated because of the combinations needed for the popup, background, and adapter pages.  Hope you like the caching :P

**Comment by paulirish** [comment]
wow yeah this is way better. :)

i dont think i can look at the build script stuff right now unfortunately

but i like what i see int he extension

will call it out soon

On Mon, Mar 12, 2012 at 11:05 PM, JJ Ford <
reply@reply.github.com
